pennylane/transforms/defer_measurements.py

```diff
@@ -104,18 +104,18 @@ def qfunc(par):
             measured_wires[op.id] = op.wires[0]
 
         elif op.__class__.__name__ == "Conditional":
-            control_wire = [measured_wires[m_id] for m_id in op.meas_val.measurements]
-            for value in op.meas_val.branches.values():
-                if value == op.meas_val.control_value:
-                    if op.meas_val.control_value == 0:
-                        qml.PauliX(Wires(control_wire))
-
-                    ctrl(
-                        lambda: apply(op.then_op),  # pylint: disable=cell-var-from-loop
-                        control=Wires(control_wire),
-                    )()
-
-                    if op.meas_val.control_value == 0:
-                        qml.PauliX(Wires(control_wire))
+            _add_control_gate(op, measured_wires)
         else:
             apply(op)
+
+
+def _add_control_gate(op, measured_wires):
+    """Helper function to add control gates"""
+    control = [measured_wires[m_id] for m_id in op.meas_val.measurement_ids]
+    for branch, value in op.meas_val._items():  # pylint: disable=protected-access
+        if value:
+            ctrl(
+                lambda: apply(op.then_op),  # pylint: disable=cell-var-from-loop
+                control=Wires(control),
```

Comment:
Note: because the PauliX filliping logic has been changed, now the tests were also adjusted because they assumed that PauliX operations are being added: https://github.com/PennyLaneAI/pennylane/pull/3159/commits/5468aeac26d58f0919f37f74f7a5438492cfd9a5

Instead, we've absorbed the flipping into the `ctrl` instruction by passing in `control_values`.

(ccing @rmoyard)

Reply:
Great thank you!